internal/providers/pluginfw/resources/registered_model/data_registered_model.go

```diff
@@ -0,0 +1,100 @@
+package registered_model
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/catalog_tf"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func DataSourceRegisteredModel() datasource.DataSource {
+	return &RegisteredModelDataSource{}
+}
+
+var _ datasource.DataSourceWithConfigure = &RegisteredModelDataSource{}
+
+type RegisteredModelDataSource struct {
+	Client *common.DatabricksClient
+}
+
+type RegisteredModelData struct {
+	FullName             types.String                    `tfsdk:"full_name"`
+	IncludeModelVersions types.Bool                      `tfsdk:"include_model_versions" tf:"optional"`
+	ModelInfo            *catalog_tf.RegisteredModelInfo `tfsdk:"model_info" tf:"optional,computed"`
+	ModelVersions        []catalog_tf.ModelVersionInfo   `tfsdk:"model_versions" tf:"optional,computed"`
```

Comment:
Does it make sense for these to be coupled together? It might make sense to have a `databricks_registered_model_versions` datasource and split this into two separate data sources, which reduces the amount of custom abstractions in the provider.

Reply:
let me look into it

Reply:
Ok, let me remove model versions and add a new data source for it.